gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -1311,6 +1311,133 @@ function plymeta:SafeDropWeapon(wep, keepSelection)
 	return true
 end
 
+local function TraceAmmoDrop(ply)
+	local pos, ang = ply:GetShootPos(), ply:EyeAngles()
+	local fwd, rgt, up = ang:Forward(), ang:Right(), ang:Up()
+
+	local dir = fwd * 32
+	rgt:Mul(6)
+	up:Mul(-5)
+	dir:Add(rgt)
+	dir:Add(up)
+
+	local tr = util.QuickTrace(pos, dir, ply)
+
+	return tr, pos, dir, fwd
+end
+
+---
+-- Checks if the ammo can be dropped in a safe manner.
+-- @param Weapon wep The weapon that should be dropped
+-- @return boolean Returns if this weapon can be dropped
+-- @realm server
+function plymeta:CanSafeDropAmmo(wep)
+	if not IsValid(self) or not (IsValid(wep) and wep.AmmoEnt) then
+		return false
+	end
+
+	local tr = TraceAmmoDrop(self)
+
+	if tr.HitWorld then
+		LANG.Msg(self, "drop_no_room_ammo", nil, MSG_MSTACK_WARN)
+
+		return false
+	end
+
+	return true
+end
+
+---
+-- Called to drop ammo in a safe manner (e.g. preparing and space-check).
+-- @param Weapon wep The weapon that should be referenced when dropping ammo.
+-- @param boolean useClip If set to true the ammo is dropped from the clip, otherwise, from reserve ammo.
+-- @return boolean Returns if ammo is dropped
+-- @realm server
+function plymeta:SafeDropAmmo(wep, useClip)
+	if not self:CanSafeDropAmmo(wep) then
+		return false
+	end
+
+	-- most of this goes into computing ammo amount
+	local amt = 0
+	if useClip then
+		amt = wep:Clip1()
+	else
+		amt = math.min(
+			wep.Primary.ClipSize,
+			self:GetAmmoCount(wep.Primary.Ammo)
+		)
+	end
+	local hook_data = {amt}
+
+	---
+	-- @realm server
+	if hook.Run("TTT2DropAmmo", self, hook_data) == false then
+		LANG.Msg(self, useClip and "drop_ammo_prevented" or "drop_reserve_prevented", nil, MSG_MSTACK_WARN)
+
+		return false
+	end
+	amt = hook_data[1]
+	--TODO: revisit here with convar for ammo quarter thing
+	if amt < 1 or amt <= wep.Primary.ClipSize * 0.25 then
+		LANG.Msg(self, useClip and "drop_no_ammo" or "drop_no_reserve", nil, MSG_MSTACK_WARN)
+
+		return false
+	end
```

Comment:
```suggestion
	if amt < 1 or amt <= wep.Primary.ClipSize * 0.25 then
		LANG.Msg(self, useClip and "drop_no_ammo" or "drop_no_reserve", nil, MSG_MSTACK_WARN)

		return false
	end
```
This should probably be an issue instead of a TODO comment. That would allow us to discuss if this restriction should be configurable at all or maybe even removed outright.